app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,103 @@
+class ProjectsController < ApplicationController
+  before_action :require_login, only: [:new, :create, :edit, :update, :destroy]
+  before_action :check_privileges, only: [:edit, :update]
+
+  def index
+    @projects = Project.all
+  end
+
+  def new
+    @project = Project.new
+  end
+
+  def create
+    @project = Project.new(project_params)
+    tags = params[:tags].split(',')
+    tags.each do |tag|
+      valid_tag = Tag.find_by(name: tag.strip.capitalize)
+      if valid_tag
+        @project.tags << valid_tag
+      else
+        flash.now[:errors] = ["The tag #{tag.strip} does not exist"]
+      end
+    end
+    if flash.now[:errors].nil? && @project.save
+      redirect_to @project
+    else
+      flash.now[:errors] = @project.errors.full_messages
+      render :new
+    end
+  end
+
+  def show
+    @project = Project.find_by(id: params[:id])
+    if @project.nil?
+      redirect_to :root, notice: "Project doesn't exist."
+    else
+      @active_collaborators = @project.collaborators_with_status("approved")
+    end
+  end
+
+  def edit
+    @project = Project.find_by(id: params[:id])
+  end
+
+  def update
+    @project = Project.find_by(id: params[:id])
+    current_tags = @project.project_tags
+    current_tags.destroy_all
+    @project.assign_attributes(project_params)
+    tags = params[:tags].split(',')
+    if @project.save
+      tags.each do |tag|
+        valid_tag = Tag.find_by(name: tag.strip.capitalize)
+        if valid_tag
+          @project.tags << valid_tag
+        else
+          flash[:notice] = "That tag does not exist"
+        end
+      end
+      redirect_to @project, notice: "Project updated successfully"
+    else
+      flash.now[:errors] = @project.errors.full_messages
+      render :edit
+    end
+  end
+
+  def destroy
+    Project.find_by(id: params[:id]).destroy
+    redirect_to root_path
+  end
+
+  def results
+    if params[:search].present?
+      @users = User.near(params[:search], params[:distance].to_i, :order => 'distance').where.not(:id => current_user.id)
+    end
+    @projects = []
+    @users.each do |user|
+      user.projects_created.each do |project|
+        @projects << project
+      end
+    end
+    if request.xhr?
+      puts "rendering project"
+      render partial: 'project', collection: @projects
+    else
+      puts "rendering not partial"
```

Comment:
You don't need this else clause - better not to have debug statements on master.